var/spack/repos/builtin/packages/qthreads/package.py

```diff
@@ -83,5 +87,51 @@ def configure_args(self):
 
         args.append('--with-scheduler=%s'
                     % self.spec.variants['scheduler'].value)
-
         return args
+
+    @run_after('install')
+    def setup_build_tests(self):
+        """Copy the build test files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        tests = self.test_list
+        relative_test_dir = self.test_base_path
+        files_to_cpy = []
+        header = 'test/argparsing.h'
+        for test in tests:
+            test_path = join_path(relative_test_dir, test + '.c')
+            files_to_cpy.append(test_path)
+        files_to_cpy.append(header)
+        self.cache_extra_test_sources(files_to_cpy)
+
+    def build_tests(self):
+        """Build and run the added smoke (install) test."""
+        tests = self.test_list
+        relative_test_dir = self.test_base_path
+
+        for test in tests:
+            options = [
+                      '-I{0}'.format(self.prefix.include),
+                      '-I{0}'.format(self.install_test_root + '/test'),
+                      join_path(self.install_test_root, relative_test_dir,
+                               test + '.c'),
+                      '-o',
+                      test,
+                      '-L{0}'.format(self.prefix.lib),
+                      '-lqthread',
+                      '{0}{1}'.format(self.compiler.cc_rpath_arg,
+                                     self.prefix.lib)]
+            reason = 'test:{0}: Checking ability to link to the library.'.format(test)
```

Comment:
You can add a slash ('\') after the apostrophe in `library.'` so Python knows to continue this statement on the next line to pick up `.format(test)`, which needs to be indented 4 spaces relative to `reason` in line 123.